src/js/components/Video/__tests__/Video-test.js

```diff
@@ -1,95 +1,174 @@
+  });
 
-// test('Video controls appear on hover', () => {
-//   const component = mount(
-//     <Grommet>
-//       <Video controls='over'>{CONTENTS}</Video>
-//     </Grommet>
-//   );
-//   console.log('!!!', component.debug());
-//   const container = component.find('StyledVideo__StyledVideoContainer');
-//   console.log('!!!', container.debug());
-//   container.simulate('mouseenter');
-//   expect(component).toMatchSnapshot();
-// });
+  test('End event handler', () => {
+    const { container } = render(
+      <Grommet>
+        <Video controls="below">{CONTENTS}</Video>
+      </Grommet>,
+    );
+    // Need to fire play event to get video playing before we fire ended event.
+    const videoContainer = document.querySelector('video');
+    fireEvent.play(videoContainer);
+    fireEvent.ended(videoContainer);
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test(' Configure Menu Button', () => {
+    window.scrollTo = jest.fn();
+    const { container, getByLabelText } = render(
+      <Grommet>
+        <Video controls="below" playing={false}>
+          {CONTENTS}
+        </Video>
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('Open Menu'));
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('fullscreen button', () => {
+    window.scrollTo = jest.fn();
+    console.warn = jest.fn();
+    const warnSpy = jest.spyOn(console, 'warn');
+    const { getByLabelText } = render(
+      <Grommet>
+        <Video controls="below" playing={false}>
+          {CONTENTS}
+        </Video>
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('Open Menu'));
+    fireEvent.click(getByLabelText('Expand'));
+    /* expect warn to have been called because jest doesn't test in any browser,
+    will always have warning here due to the jest browser not supporting
+     fullscreen */
+    expect(warnSpy).toHaveBeenCalledWith(
+      "This browser doesn't support fullscreen.",
+    );
+    warnSpy.mockReset();
+    warnSpy.mockRestore();
+    console.warn.mockRestore();
+    window.scrollTo.mockRestore();
+  });
+
+  test('play button', () => {
+    const playStub = jest
+      .spyOn(window.HTMLMediaElement.prototype, 'play')
+      .mockImplementation(() => {});
+    const { getByLabelText } = render(
+      <Grommet>
+        <Video controls="below" playing={false}>
+          {CONTENTS}
+        </Video>
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('Play'));
+    expect(playStub).toHaveBeenCalled();
+    playStub.mockRestore();
+  });
+
+  test('volume controls', () => {
+    const volMock = jest.fn();
+    window.scrollTo = jest.fn();
+    const { getByLabelText } = render(
+      <Grommet>
+        <Video controls="below" playing={false} onVolumeChange={volMock}>
+          {CONTENTS}
+        </Video>
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('Open Menu'));
+    fireEvent.click(getByLabelText('VolumeLow'));
+    expect(volMock).toHaveBeenCalled();
```

Comment:
Can we do the same for raising the volume?

Reply:
At first it appeared that there was a bug because the volumeUp button would no fire the `onVolumeChange` even, but it looks like the volume starts out at its maximum so if you lower the volume first you can then raise it. The test for the volume Up is included in the latest commit